helix-term/src/commands/typed.rs

```diff
@@ -2142,6 +2142,53 @@ fn reflow(
     Ok(())
 }
 
+// This is a basic implementation for viewing the entire tree sitter tree of a file
+// In the future, this command could be updated such that it opens a buffer which automatically highlights the tree sitter sub-tree of where we currently have our cursor
+// And vice-versa: Moving cursor across nodes in the tree sitter tree will need to also move the cursor across the original file
+fn tree_sitter_tree(
+    cx: &mut compositor::Context,
+    _args: &[Cow<str>],
+    event: PromptEvent,
+) -> anyhow::Result<()> {
+    if event != PromptEvent::Validate {
+        return Ok(());
+    }
+
+    // Safety: 100_000_000 != 0
+    let doc_id =
+        DocumentId::new(unsafe { NonZeroUsize::new_unchecked(TREE_SITTER_TREE_DOCUMENT_ID) });
```

Comment:
This is exposing all sorts of implementation details and we certainly don't want to expose document ID creation or allowing to specify the ID manually (and potentially overlap an existing doc).  Why not just use a regular buffer? There's no need for it to have a special ID

Reply:
Okay, so I changed the implementation to instead store an `Option` on the editor that stores the ID of the tree sitter document

From within the `tree_sitter_tree`, I need to know whether a `tree_sitter_tree` is already opened so I can replace it with a new one. I've looked at all of the `Document` APIs and it seems that I need to store a global reference to that Document ID.

If there's an alternative better solution, please let me know!

Reply:
I'll mark it as "resolved" since I've changed the implementation to not expose document ID creation